src/javascript/_common/geocoder.js

```diff
@@ -0,0 +1,48 @@
+/* global google */
+const scriptjs = require('scriptjs');
+const Client   = require('../app/base/client');
+const localize = require('../_common/localize').localize;
+
+const Geocoder = (() => {
+    const validate = (address, residence) => (
+        new Promise((resolve) => {
+            scriptjs('https://maps.googleapis.com/maps/api/js?key=AIzaSyAEha6-HeZuI95L9JWmX3m6o-AxQr_oFqU&libraries=places', () => {
+                const country  = Client.get('residence') || residence;
+                const geocoder = new google.maps.Geocoder();
+                geocoder.geocode({
+                    address,
+                    componentRestrictions: { country },
+                }, (result, status) => {
+                    // Geocoding status reference:
+                    // https://developers.google.com/maps/documentation/javascript/geocoding#GeocodingStatusCodes
+                    handleResponse(status);
+                    resolve(status);
+                });
+            });
+        })
+    );
+
+    const handleResponse = (status) => {
+        const $geocode_error = $('#geocode-error');
+        if (status === google.maps.GeocoderStatus.ZERO_RESULTS) {
+            if ($geocode_error.length) {
+                $geocode_error.fadeIn(0);
+                return;
+            }
+            const $last_child = $('#address_form').children(':last');
+            $last_child.parent().append($('<p/>', {
+                id   : 'geocode-error',
+                class: 'notice-msg no-margin',
+                text : localize('Your address could not be verified by our automated system. You may proceed but please ensure that your address is complete.'),
+            }));
+        } else if (status === google.maps.GeocoderStatus.OK) {
```

Comment:
Do we need to handle other _Status Codes_ as well?

i.e. if it returns `ZERO_RESULTS` first, then on client changes the address to a correct one but it returns `OVER_QUERY_LIMIT` or other error codes, the above message shown on the first try, won't be disappeared.